src/bundles/analytics.test.js

```diff
@@ -27,152 +27,169 @@ function createStore (analyticsOpts = {}) {
-  global.Countly.opt_in.mockClear()
-  global.Countly.opt_out.mockClear()
-
-  store.doToggleAnalytics()
-  expect(store.selectAnalyticsEnabled()).toBe(false)
-  expect(store.selectAnalyticsConsent()).toEqual([])
-  expect(global.Countly.opt_in).not.toHaveBeenCalled()
-  expect(global.Countly.opt_out).toHaveBeenCalled()
-  expect(global.Countly.opt_out.mock.calls.length).toBe(1)
-  global.Countly.opt_in.mockClear()
-  global.Countly.opt_out.mockClear()
-
-  store.doToggleAnalytics()
-  expect(store.selectAnalyticsEnabled()).toBe(true)
-  expect(store.selectAnalyticsConsent()).toEqual(['sessions', 'events', 'views', 'location'])
-  expect(global.Countly.opt_in).toHaveBeenCalled()
-  expect(global.Countly.opt_out).not.toHaveBeenCalled()
-  expect(global.Countly.opt_in.mock.calls.length).toBe(1)
-  global.Countly.opt_in.mockClear()
-  global.Countly.opt_out.mockClear()
-
-  store.doToggleConsent('sessions')
-  expect(store.selectAnalyticsEnabled()).toBe(true)
-  expect(store.selectAnalyticsConsent()).toEqual(['events', 'views', 'location'])
-  // changing the consents should not opt/in or out
-  expect(global.Countly.opt_in).not.toHaveBeenCalled()
-  expect(global.Countly.opt_out).not.toHaveBeenCalled()
-  // reset the mocks here to simplify the following assetions
-  global.Countly.opt_in.mockClear()
-  global.Countly.opt_out.mockClear()
-
-  store.doToggleConsent('location')
-  expect(store.selectAnalyticsEnabled()).toBe(true)
-  expect(store.selectAnalyticsConsent()).toEqual(['events', 'views'])
-  // changing the consents should not opt/in or out
-  expect(global.Countly.opt_in).not.toHaveBeenCalled()
-  expect(global.Countly.opt_out).not.toHaveBeenCalled()
-  // reset the mocks here to simplify the following assetions
-  global.Countly.opt_in.mockClear()
-  global.Countly.opt_out.mockClear()
-
-  store.doToggleConsent('views')
-  expect(store.selectAnalyticsEnabled()).toBe(true)
-  expect(store.selectAnalyticsConsent()).toEqual(['events'])
-  // changing the consents should not opt/in or out
-  expect(global.Countly.opt_in).not.toHaveBeenCalled()
-  expect(global.Countly.opt_out).not.toHaveBeenCalled()
-  // reset the mocks here to simplify the following assetions
-  global.Countly.opt_in.mockClear()
-  global.Countly.opt_out.mockClear()
-
-  // Removing all consent is equivalent to disabling the analytics.
-  store.doToggleConsent('events')
-  expect(store.selectAnalyticsEnabled()).toBe(false)
-  expect(store.selectAnalyticsConsent()).toEqual([])
-  expect(global.Countly.opt_in).not.toHaveBeenCalled()
-  expect(global.Countly.opt_out).toHaveBeenCalled()
-  expect(global.Countly.opt_out.mock.calls.length).toBe(1)
-  global.Countly.opt_in.mockClear()
-  global.Countly.opt_out.mockClear()
+describe('user manages analytics consent with individual settings toggles', () => {
+  it('should toggle consent on for "crashes"', () => {
+    const mockDefaultState = {
+      lastEnabledAt: Date.now,
+      lastDisabledAt: 1674166495717, // past
```

Comment:
use explicit dates for tests to make things easier for us in the future.

```suggestion
      lastEnabledAt: (new Date('2022-01-02')).getTime(),
      lastDisabledAt: (new Date('2022-01-01')).getTime(), // past
```